c_src/erlzmq_nif.c

```diff
@@ -920,6 +992,66 @@ SOCKET_COMMAND(erlzmq_socket_command_recv)
   return result;
 }
 
+SOCKET_COMMAND(erlzmq_socket_command_recv_multipart)
+{
+  assert(argc == 1);
+  int flags;
+
+  if (! enif_get_int(env, argv[0], &flags)) {
+    return enif_make_badarg(env);
+  }
+
+  assert(socket->socket_zmq);
+
+  ERL_NIF_TERM list = enif_make_list_from_array(env, NULL, 0);
+
+  for (int i = 0;;) {
+    zmq_msg_t msg;
+    if (zmq_msg_init(&msg)) {
+      return return_zmq_errno(env, zmq_errno());
+    }
+
+    if (zmq_recvmsg(socket->socket_zmq, &msg, flags) == -1) {
+      if (errno == EINTR && i > 0) {
```

Comment:
I'm not sure it can happen. AFAIK ZMQ guarantees atomic read of multipart messages. It means only on the first one it can sleep and be interrupted. Besides you should use zmq_errno() here.

Reply:
I'm not sure using zmq_errno is correct, will check. See https://github.com/zeromq/czmq/blob/master/src/zmsg.c#L117-L153

Reply:
The EINTR checking is directly derived from the code in czmq. I would rather not rely on making the assumption that every call after the first successful call can not be interrupted.

Reply:
> I'm not sure using zmq_errno is correct, will check. See https://github.com/zeromq/czmq/blob/master/src/zmsg.c#L76-L114

It's correct but not necessary on POSIX
> The zmq_errno() function is provided to assist users on non-POSIX systems who are experiencing issues with retrieving the correct value of errno directly

Reply:
> The EINTR checking is directly derived from the code in czmq. I would rather not rely on making the assumption that every call after the first successful call can not be interrupted.

I guess it's battle tested then

Reply:
made the errno change